BoogieLang/Ast_Cfg_Transformation.thy

```diff
@@ -0,0 +1,2280 @@
+next
+  case 1: (Suc l1)
+  then show ?thesis 
+    proof (cases l1)
+      case 0
+      from 1 assms this have 
+        "(red_bigblock A M \<Lambda> \<Gamma> \<Omega> T (BigBlock None [] None None, KEndBlock (KSeq (BigBlock name [] (Some (WhileWrapper str)) tr) cont0), Normal ns3) (reached_bb, reached_cont, reached_state))"
+        by fastforce
+      then show ?thesis
+      proof cases
+        case RedSkipEndBlock thus ?thesis by (simp add: Ast.valid_configuration_def)
+      qed auto
+    next
+      case 2: (Suc l2)
+      then show ?thesis
+      proof (cases l2)
+        case 0 
+        from 2 1 have "l = Suc (Suc l2)" by auto
+        from 2 1 assms obtain inter_bb inter_cont inter_state  where
+          step1: "(red_bigblock A M \<Lambda> \<Gamma> \<Omega> T (BigBlock None [] None None, KEndBlock (KSeq (BigBlock name [] (Some (WhileWrapper str)) tr) cont0), Normal ns3) (inter_bb, inter_cont, inter_state))" and 
+          step2: "(red_bigblock A M \<Lambda> \<Gamma> \<Omega> T (inter_bb, inter_cont, inter_state) (reached_bb, reached_cont, reached_state))" 
+          using "0" by auto
+        from this step1 have "(inter_bb, inter_cont, inter_state) = (BigBlock None [] None None, (KSeq (BigBlock name [] (Some (WhileWrapper str)) tr) cont0), Normal ns3)"
+        proof cases
+          case RedSkipEndBlock thus ?thesis 
+            by blast
+        qed auto
+        from step2 this have "(reached_bb, reached_cont, reached_state) = ((BigBlock name [] (Some (WhileWrapper str)) tr), cont0, Normal ns3)" 
+        proof cases 
+          case RedSkip thus ?thesis  using \<open>(inter_bb, inter_cont, inter_state) = (BigBlock None [] None None, (KSeq (BigBlock name [] (Some (WhileWrapper str)) tr) cont0), Normal ns3)\<close> by fastforce
+        qed auto
+        then show ?thesis by (simp add: Ast.valid_configuration_def)
+      next
+        case 3: (Suc l3)
+        from 3 2 1 have "l = Suc (Suc (Suc l3))" by auto
+        from 3 2 1 assms obtain inter_bb inter_cont inter_state inter_bb2 inter_cont2 inter_state2 inter_bb3 inter_cont3 inter_state3 where
+          step1: "(red_bigblock A M \<Lambda> \<Gamma> \<Omega> T (BigBlock None [] None None, KEndBlock (KSeq (BigBlock name [] (Some (WhileWrapper str)) tr) cont0), Normal ns3) (inter_bb, inter_cont, inter_state))" and 
+          step2: "(red_bigblock A M \<Lambda> \<Gamma> \<Omega> T (inter_bb, inter_cont, inter_state) (inter_bb2, inter_cont2, inter_state2))" and 
+          step3: "(red_bigblock A M \<Lambda> \<Gamma> \<Omega> T (inter_bb2, inter_cont2, inter_state2) (inter_bb3, inter_cont3, inter_state3))" and
+          rest: "A,M,\<Lambda>,\<Gamma>,\<Omega>,T \<turnstile> (inter_bb3, inter_cont3, inter_state3) -n\<longrightarrow>^l3 (reached_bb, reached_cont, reached_state)" 
+          by (metis (no_types, opaque_lifting) get_state.cases relpowp_Suc_D2)
```

Comment:
replace by "by (metis (no_types) etc...)", otherwise the proofs fail on Windows.